sdk/src/main/java/com/hedera/hashgraph/sdk/Network.java

```diff
@@ -159,167 +130,94 @@ NodeAddress getAddressBook(AccountId nodeAccountId) {
+        var nodes = getNumberOfMostHealthyNodes(getNumberOfNodesForRequest());
+        var nodeAccountIds = new ArrayList<AccountId>(nodes.size());
 
-        List<AccountId> resultNodeAccountIds = new ArrayList<>();
-
-        for (int i = 0; i < getNumberOfNodesForTransaction(); i++) {
-            resultNodeAccountIds.add(nodes.get(i).accountId);
+        for (var node : nodes) {
+            nodeAccountIds.add(node.getAccountId());
         }
 
-        lock.release();
-
-        return resultNodeAccountIds;
-    }
-
-    void setMaxNodesPerTransaction(int maxNodesPerTransaction) {
-        this.maxNodesPerTransaction = maxNodesPerTransaction;
+        return nodeAccountIds;
     }
 
-    int getMaxNodeAttempts() {
-        return maxNodeAttempts;
-    }
-
-    void setMaxNodeAttempts(int maxNodeAttempts) {
-        this.maxNodeAttempts = maxNodeAttempts;
-    }
-
-    Duration getNodeWaitTime() {
-        return nodeWaitTime;
-    }
-
-    void setNodeWaitTime(Duration nodeWaitTime) {
-        this.nodeWaitTime = nodeWaitTime;
-        for (var node : nodes) {
-            node.setWaitTime(nodeWaitTime.toMillis());
-        }
+    public Network setMaxNodesPerRequest(int maxNodesPerRequest) {
+        this.maxNodesPerRequest = maxNodesPerRequest;
+        return this;
     }
 
-    int getNumberOfNodesForTransaction() {
-        if (maxNodesPerTransaction != null) {
-            return Math.min(maxNodesPerTransaction, nodes.size());
+    public int getNumberOfNodesForRequest() {
+        if (maxNodesPerRequest != null) {
+            return Math.min(maxNodesPerRequest, nodes.size());
         } else {
             return (nodes.size() + 3 - 1) / 3;
         }
     }
 
-    void close(Duration timeout) throws TimeoutException {
-        try {
-            lock.acquire();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
-
-        var stopAt = Instant.now().getEpochSecond() + timeout.getSeconds();
-
-        for (var node : nodes) {
-            if (node.channel != null) {
-                node.channel = node.channel.shutdown();
-            }
-        }
-
-        for (var node : nodes) {
-            if (stopAt - Instant.now().getEpochSecond() == 0) {
-                lock.release();
-                throw new TimeoutException("Failed to properly shutdown all channels");
-            }
-
-            if (node.channel != null) {
-                try {
-                    node.channel.awaitTermination(stopAt - Instant.now().getEpochSecond(), TimeUnit.SECONDS);
-                } catch (InterruptedException e) {
-                    lock.release();
-                    throw new RuntimeException(e);
-                }
-            }
-        }
-
-        nodes.clear();
-        networkNodes.clear();
-        network.clear();
-
-        lock.release();
+    public Node getNodeForAccountId(AccountId nodeAccountId) {
```

Comment:
The `ForAccountId` is implied by the parameter and can be omitted.

```suggestion
    public Node getNode(AccountId nodeAccountId) {
```

Reply:
Fair point, but the reason I added `ForAccountId` is because I expect any `getX` methods to not take any parameters as that is the Java convention, no?

Reply:
The Java beans standards says that for exposing fields you should name the getter as `getFieldName()`. You're not exposing a field here but are performing a lookup of information so it can be named whatever you like. Omitting the `For` part allows for overriding of the method with different views indexed by a different parameter. For example `getNode(String endpoint)`. It's not a big deal here with just one parameter, but It's really bad when there are multiple parameters and people duplicate the parameter names in the method name (e.g. `getNodeByAccountIdAndEndpoint(AccountId accountId, String endpoint)`.